core/src/Kokkos_OpenMP.hpp

```diff
@@ -182,14 +182,22 @@ class OpenMP {
   static int impl_get_current_max_threads() noexcept;
 
   Impl::OpenMPInternal* impl_internal_space_instance() const {
+#ifdef KOKKOS_IMPL_WORKAROUND_ICE_IN_TRILINOS_WITH_OLD_INTEL_COMPILERS
+    return m_space_instance;
+#else
     return m_space_instance.get();
+#endif
   }
 
   static constexpr const char* name() noexcept { return "OpenMP"; }
   uint32_t impl_instance_id() const noexcept { return 1; }
 
  private:
+#ifdef KOKKOS_IMPL_WORKAROUND_ICE_IN_TRILINOS_WITH_OLD_INTEL_COMPILERS
+  Impl::OpenMPInternal* m_space_instance;
```

Comment:
Could we use a regular shared_ptr instead? Then we would not need to change the rest of the code.

Reply:
I tried testing diff below with intel/19.0.5 using release-candidate-3.7.00 (instead of changes in this PR), but ICE still occurs:

```diff
diff --git a/core/src/Kokkos_OpenMP.hpp b/core/src/Kokkos_OpenMP.hpp
index 38a537b..74ca3d5 100644
--- a/core/src/Kokkos_OpenMP.hpp
+++ b/core/src/Kokkos_OpenMP.hpp
@@ -189,7 +189,7 @@ class OpenMP {
   uint32_t impl_instance_id() const noexcept { return 1; }
 
  private:
-  Kokkos::Impl::HostSharedPtr<Impl::OpenMPInternal> m_space_instance;
+  std::shared_ptr<Impl::OpenMPInternal> m_space_instance;
 };
```